### PR TITLE
Initial stab at bash completion blog post

### PR DESCRIPTION
**Pull request by Crisfole**
~This is a first stab. I haven't even tested to see if everything looks right yet. Just getting content out there.~

This is about as polished as I usually get my writing before showing it to other people.

### CODE REVIEW

**Comment by egreer** [comment]
Worked on my machine. Thanks Chris. Post seems fine to me.

File: _drafts/bash_completion.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+Bash Programmable Autocompletion is powerful, but intimidating at first. Hopefully this post helps you get started.
+If you build something awesome write it up and ping us on twitter! We're [@ApsisLabs](https://twitter.com/apsislabs).
+
+
+[^1]: Couldn't use an alias, because aliases can't interpolate arguments, they're strictly dumb text replacements , and they always include a space at the end.
+
+[^2]: [xkcd is relevant here](https://xkcd.com/1205/)
+
+[^3]: The bash man pages use tons of bolded references to dozens of other locations in a what is a huge doc with no links. This is becuase it's a man page, and man pages, as useful as they are, are dinosaurs [from a prehistoric time before hypertext was invented](https://unix.stackexchange.com/a/18161/34182).
```
Comment (HenryKeiter) [inline _drafts/bash_completion.md]: "dozens of other locations in a what is a huge doc with no links." You've got a few too many articles/prepositions in here.

File: _drafts/bash_completion.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
+---
+layout: post
+author: chris
+title: Bash Directory Completion
```
Comment (nic-at-apsis) [inline _drafts/bash_completion.md]: Maybe title it something like: Save Time Through Bash Directory Autocompletion?

File: _drafts/bash_completion.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+excerpt: >
+    What follows is a short tutorial on bash autocompletion and a tiny bit of bash programming
+    information. It assumes you have a working knowledge of programming, and at least passing
+    familiarity with your terminal.
+---
+
+Every developer should have at least a basic mastery of their command line. I'd argue
+that part of that mastery is to develop a set of customizations they take with them
+wherever they go. These customizations should make things you do all the time faster.
+Obviously there are diminshing returns here[^2].
```
Comment (nic-at-apsis) [inline _drafts/bash_completion.md]: does footnote #2 show up first? maybe switch numbering if so...

File: _drafts/bash_completion.md (hunk 2 above)
Comment (Crisfole) [reply _drafts/bash_completion.md]: Done. "Save time through bash programmable autocompletion"

**Pull request merged**